### DISCUSSION TITLE
Switch on and  off from SHM

### DISCUSSION THREAD

**Discussion by Rinusesa** [camueller/SmartApplianceEnabler#182] [Allgemein]
Dear Alex 

is it possible to configure a device so it works like the edimax plug? I would like to be able to switch a Solid state relay connected to GPIO , On and Off from SHM. Now I can only create a device but I am not able to switch on and off in SHM…

**Comment by camueller** [comment]
You cannot control a SEMP device from the Sunny Portal. But you can control it using the SAE web interface. SAE can control a SSR via GPIO - I'm running this setup myself.

**Reply by Rinusesa** [reply]
I have set up the raspberry pi with SAE today. I was able to set it up in SAE and switch and measure a device. I am used working with sunny portal and do the setup there. I can also do the configuration online ( not in the same network ). To setup the SAE I need to be in the same network I guess?

**Reply by camueller** [reply]
Yes, the SEMP protocol only works if SHM and SAE are in the same subnet.

**Reply by Rinusesa** [reply]
Possible to instal VNc on the raspberry? Great programming by the way 👍

**Reply by camueller** [reply]
I prefer working with a shell but why shouldn't you be able to set up VNC?

BTW. Where are you from?

**Reply by Rinusesa** [reply]
I am from Belgium. I use the SHM but could not measure 3 phase. Now thx to u I can :)

**Reply by camueller** [reply]
Some months ago I started to provide the web interface in english but documentation is still available in German ony. I guess that should be no problem with Google Translate. AFAIK you are the second non-german user of the SAE - the other one lives in Portugal. In the furture I might provide documentation in english but not just for 2 users :grin:

**Reply by Rinusesa** [reply]
It's a really nice project ! I have a powerdale Nexxtender home EV charger. Can this one be integrated in the SAE as well?

**Reply by camueller** [reply]
If it can be controlled via HTTP oder Modbus. I would reocmmend to select a Wallbox from the templates having the same protocol like your Wallbox. The settings derived from the template should give you an idea what you have to look for in the API spec of your wallbox. Then you simply change the settings as you like and test if it is working. Maybe you should open a separate discussion for this subject ...

**Reply by Rinusesa** [reply]
Thx i check it out. And can't u add a device in the software so it acts like an edimax plug ? Controllable through SHM instead of SAE? Maybe in an update ?

**Reply by camueller** [reply]
No. As I already stated: It is not possible because of SEMP protocol.

**Comment by Rinusesa** [comment]
Hey. I figured it out and works fine. But is there a way I can connect to the raspberry pi remotely ( through internet ) so I can configure the SAE? When I am in the same network it's no problem, but when i am elsewhere I can't connect. I tried vnc but it is not displaying anything. I need to be able to connect to the raspberry pi and get in the same network. But I don't know how :)

**Comment by solanjo** [comment]
Maybe you can establish a vpn connection. I use this from my mobile and it works fine. Or you could open a dedicated port on your internet router.

**Comment by camueller** [comment]
You only have to forward a port from your router to port 8080 on the raspberry pi. However then anyone can configure your SAE from the internet since SAE doesn't have any authentication features. Therefore I would not forward to port 8080 on the reaspberry pi but to port 22 in order to use SSH. You may tunnel anything through SSH including the SAE web interface running on port 8080.

**Reply by Rinusesa** [reply]
Thx, now I know that it is possible. But I don't know anything about ssh tunneling and I have a lot to google... 

I guess I need to set up ssh on the raspberry pi through sudo raspi-config and then make a connection from my computer with putty 

But how can I then get to the SAE? Just in the browser ? And this can only work if I open up the port 22 on the router

**Comment by camueller** [comment]
You can use any port you like on the router (e.g. 1234) and forward it to port 22 on the raspberry pi. You should google for something like "port forwarding through ssh" - this is a general topic not related to SAE.

**Comment by Rinusesa** [comment]
Hey, one more thing. If i configure a device ( pool filterpump ) like in the picture. The pump CAN run for max 6 hours a day in the fime frame 08-20:00 yes ? If i look in the sunny home manager i can see that the filterpump is dark red ( in my opinion that means that it will run ) while there is no sun. did i configure something wrong? 

![image](https://user-images.githubusercontent.com/98557292/152931336-02227bfc-ad20-4b4b-a3ce-60c2f7490674.png)

![image](https://user-images.githubusercontent.com/98557292/152931287-5af63b42-ec71-43fd-b8fd-03c435c066d4.png)

![image](https://user-images.githubusercontent.com/98557292/152931220-e57e9500-eddd-4437-ba6a-8f64088c429b.png)

**Comment by camueller** [comment]
If it should only run on excess energy you have to enter "0:00" for "Min. runtime". If you enter "03:00" it will run at least 3 hours but may run up to 6 hours if there is excess energy available. Perhaps I should add a tooltip to these fields ...

**Reply by Rinusesa** [reply]
Yes that's what i thought but the Home manager plans it like it has to run... it affects other devices ... normally if it runs only with excess energy it shows in a light color in the Home manager. Mine shows dark color for 6 hours...

**Comment by camueller** [comment]
If you enter a value for "Min. runtime" the Home Manger will display it with a light color after some minutes.

**Reply by Rinusesa** [reply]
Aha, thx. I did not enter anything! That will be it. Thx a lot

**Reply by Rinusesa** [reply]
Hey, just wanted to le u know that now it works very nice ! thx for the support !

maybe one more question , i can see in the sunnyportal that my pump has used 24kwh in total , but on the energymeter itself i have 32kwh. thats a big difference. u think the difference is because i tried with a very cheap energy meter or the difference is normal? it has a 1000 impulse output so thats how i configured it in the smart appliance enabler with a 15ms minimum pulse duration

friendly regards

verbeke rinus

**Comment by camueller** [comment]
In SAE you should configure the HTTP meter parameter for "Meter reading". This provided the most accurate reading of power consumption.

**Reply by Rinusesa** [reply]
What https energy meter are u using ?

**Comment by camueller** [comment]
I'm using various devices (Nouse A1, Gosund SP111, Sonoff Pow) all flashed with Tasmota.
